File: billing-domain/src/main/java/com/example/billing/domain/credit/InsufficientCreditException.java
package com.example.billing.domain.credit;

import com.example.billing.domain.shared.Money;

public class InsufficientCreditException extends RuntimeException {

    private final CreditId creditId;
    private final Money requested;
    private final Money available;

    public InsufficientCreditException(CreditId creditId, Money requested, Money available) {
        super("insufficient credit: id=" + creditId + " requested=" + requested + " available=" + available);
        this.creditId = creditId;
        this.requested = requested;
        this.available = available;
    }

    public CreditId creditId() { return creditId; }
    public Money requested() { return requested; }
    public Money available() { return available; }
}

File: billing-domain/src/main/kotlin/com/example/billing/domain/credit/InsufficientCreditException.kt
package com.example.billing.domain.credit

import com.example.billing.domain.shared.Money

/**
 * 크레딧 차감 시 잔액 부족.
 *
 * record-style accessor (`creditId()`, `requested()`, `available()`) 는 `@get:JvmName` 으로
 * Java/Kotlin 양쪽 호출자에서 그대로 호출 가능.
 */
class InsufficientCreditException(
    @get:JvmName("creditId") val creditId: CreditId,
    @get:JvmName("requested") val requested: Money,
    @get:JvmName("available") val available: Money,
) : RuntimeException("insufficient credit: id=$creditId requested=$requested available=$available")
